src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -356,13 +356,13 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
      * @throws CommCareSessionException
      */
     private Screen autoAdvanceSession(
+            @Nullable Screen nextScreen,
```

Comment:
nit: this naming seems confusing, isn't it actually the current screen we want to pass in?

This is also only relevant for EOF right? I would be a bit confused on when this should or shouldn't be null when using the function. Maybe a docstring would help?

Reply:
Added [here](https://github.com/dimagi/formplayer/pull/1620/commits/6e5ce3d44e9002877201ccbaa8e84c72a981a624).

> This is also only relevant for EOF right? 

Not really, it can be used in another navigation pathways as well if the screen is already evaluated for the session. The other pathway where it's passed as `null` in the PR doesn't have the screen evaluated already and therefore we need it to re-calculate as part of this method. 

> I would be a bit confused on when this should or shouldn't be null when using the function.

Agree it's a bit tricky as one will need to check whether we have already evaluated the screen as per the current session state before calling this.